modules/core/02-client/keeper/grpc_query.go

```diff
@@ -240,3 +234,40 @@ func (q Keeper) UpgradedClientState(c context.Context, req *types.QueryUpgradedC
 		UpgradedClientState: any,
 	}, nil
 }
+
+// UpgradedConsensusState implements the Query/UpgradedConsensusState gRPC method
+func (q Keeper) UpgradedConsensusState(c context.Context, req *types.QueryUpgradedConsensusStateRequest) (*types.QueryUpgradedConsensusStateResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+
+	plan, found := q.GetUpgradePlan(ctx)
+	if !found {
+		return nil, status.Error(
+			codes.NotFound, "upgrade plan not found",
+		)
+	}
+
+	bz, found := q.upgradeKeeper.GetUpgradedConsensusState(ctx, plan.Height)
```

Comment:
This can simply return early by checking plan.Height rather than doing a store query